hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.test.JetAssert;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.impl.util.IOUtil.packDirectoryIntoZip;
+import static com.hazelcast.jet.impl.util.IOUtil.unzip;
+
+@RunWith(HazelcastParallelClassRunner.class)
+public class IOUtilTest extends JetTestSupport {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void test_zipAndUnzipNestedFolder_then_contentsShouldBeSame() throws Exception {
+        //Given
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(1000);
+
+        //When
+        Path originalPath = Paths.get(this.getClass().getResource("/nested").toURI());
```

Comment:
IIUIC, git also doesn't store empty directories. If you had an empty directory in `/nested`, they aren't there when you check out from git.

Reply:
Nope, there wasn't an empty directory under `/nested` folder

Reply:
We should test empty directories though